executor/joiner.go

```diff
@@ -433,5 +631,39 @@ func (j *innerJoiner) tryToMatch(outer chunk.Row, inners chunk.Iterator, chk *ch
 
 }
 
-func (j *innerJoiner) onMissMatch(outer chunk.Row, chk *chunk.Chunk) {
+func (j *innerJoiner) onMissMatchInners(outer chunk.Row, chk *chunk.Chunk) {
+}
+
+// tryToMatchOuters implements joiner interface.
+func (j *innerJoiner) tryToMatchOuters(outer chunk.Row, inners chunk.Iterator, chk *chunk.Chunk) (bool, error) {
```

Comment:
Seems all `tryToMatchOuters` are mostly the same as their corresponding `tryToMatchInners`.

So how about introduce a help function like 
`tryToMatch(chunk.Row, chunk.Iterator,  *chunk.Chunk, innerFlag bool)`, 
and let both `tryToMatchOuters` and `tryToMatchInners` call it?

Then the number of lines of this file can decrease and 
it's better for maintaining since you only have to modify one place if you want to change the match function.